nufi-official/nufi-dapp-sdk
# Add GH action to sync docs with gitbook repo

### Motivation
Currently, gitbook is synchronized with https://github.com/nufi-official/nufi.

We aim to integrate content from this repository into GitBook. However, GitBook currently doesn't support syncing with multiple repositories. To avoid the hassle of updating content in two separate locations, this PR introduces a GitHub Actions workflow that:

- watches git 'pushes' on specific branch for specific files
- copies (whole) changed files into https://github.com/nufi-official/nufi repository into specific folder with specific name
- changes can be sync only as 'copying and pasting' of entire file

### How to test

1. checkout to this branch
2. make changes in `docs/cardanoNuFiConnect.md` or `docs/cardanoSnap.md`
3. commit &. push changes
4. Your changes should trigger action like this: https://github.com/nufi-official/nufi-dapp-sdk/actions/runs/9084537929
5. Consequently, changes should be visible in https://github.com/nufi-official/nufi/ on [this branch](https://github.com/nufi-official/nufi/commits/nufi-dapp-sdk-readme-sync-test/)
<img width="1316" alt="image" src="https://github.com/nufi-official/nufi-dapp-sdk/assets/64598949/2872e356-9141-4b5b-863e-4fe42ed872a8">
